### DISCUSSION TITLE
Optimize when loading over 800M GLB File

### DISCUSSION THREAD

**Discussion by sunyonglincanada** [visgl/deck.gl#5305] [Q&A] [answered]
Hi,

I am trying to load one glb file using GLTF loader which is over 800MB, I've found it is kinda stuck when loading such a big file. I am wondering if is there any way to optimize? Thanks in advance!!

**Comment by ibgreen** [comment, marked as answer]
We haven't tested with any files that big. 

There is the loading itself, which is done by loaders.gl and from your description it seems like it is freezing at this stage. 

But even if that completes then the rendering of the loaded data might be slow if the geometry is really big.

Does this file load and render well in other software?

Can you provide a public URL to the file so that we can test and see if there are any obvious inefficiencies?

**Reply by sunyonglincanada** [reply]
> We haven't tested with any files that big.
> 
> There is the loading itself, which is done by loaders.gl and from your description it seems like it is freezing at this stage.
> 
> But even if that completes then the rendering of the loaded data might be slow if the geometry is really big.
> 
> Does this file load and render well in other software?
> 
> Can you provide a public URL to the file so that we can test and see if there are any obvious inefficiencies?

Thanks for the updates!! 

Here is the public URL to the file - https://resource-1300645485.cos.ap-chengdu.myqcloud.com/model/test/factory.glb

And I've tried this online viewer - https://gltf-viewer.donmccurdy.com/ which renders the file well but it is based on the three.js.

**Reply by ibgreen** [reply]
https://github.com/visgl/loaders.gl/issues/1082

**Reply by sunyonglincanada** [reply]
> visgl/loaders.gl#1082

Thanks!!

**Reply by ibgreen** [reply]
The loaders.gl issue is updated. 

`Downloading the file from the server is slow. It takes a few minutes to download it. https://gltf-viewer.donmccurdy.com/ works with local files so it doesn't waste time on downloading.
I checked loading of this file with "loaders.gl" from local file system. It works approximately with the same speed.`

Recommend further discussions in loaders.gl issue, but unless additional info is provided we will close this.
